packages/block-library/src/social-link/edit.js

```diff
@@ -27,34 +32,51 @@ const SocialLinkEdit = ( { attributes, setAttributes, isSelected } ) => {
 
 	// Import icon.
 	const IconComponent = getIconBySite( site );
+	const socialLinkName = getNameBySite( site );
 
 	return (
-		<Button
-			className={ classes }
-			onClick={ () => setPopover( true ) }
-		>
-			<IconComponent />
-			{ isSelected && showURLPopover && (
-				<URLPopover
-					onClose={ () => setPopover( false ) }
-				>
-					<form
-						className="block-editor-url-popover__link-editor"
-						onSubmit={ ( event ) => {
-							event.preventDefault();
-							setPopover( false );
-						} } >
-						<URLInput
-							value={ url }
-							onChange={ ( nextURL ) => setAttributes( { url: nextURL } ) }
-							placeholder={ __( 'Enter Address' ) }
-							disableSuggestions={ true }
+		<Fragment>
+			<InspectorControls>
+				<PanelBody title={ sprintf( __( '%s Label' ), socialLinkName ) } initialOpen={ false }>
+					<PanelRow>
+						<TextControl
+							label={ __( 'Link Label' ) }
+							help={ __( 'Briefly describe the link to help screen reader users.' ) }
+							value={ label }
+							onChange={ ( value ) => setAttributes( { label: value } ) }
 						/>
-						<Button icon="editor-break" label={ __( 'Apply' ) } type="submit" />
-					</form>
-				</URLPopover>
-			) }
-		</Button>
+					</PanelRow>
+				</PanelBody>
+			</InspectorControls>
+			<Button
+				className={ classes }
+				onClick={ () => setPopover( true ) }
+			>
+				<IconComponent />
+				{ isSelected && showURLPopover && (
+					<URLPopover
+						onClose={ () => setPopover( false ) }
+					>
+						<form
+							className="block-editor-url-popover__link-editor"
+							onSubmit={ ( event ) => {
+								event.preventDefault();
+								setPopover( false );
+							} } >
+							<div className="editor-url-input block-editor-url-input">
```

Comment:
This is not a valid class name for this component:

https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#naming

It violates the guideline to not inherit styles from other compoennts:

>A component's class name should never be used outside its own folder (with rare exceptions such as _z-index.scss). If you need to inherit styles of another component in your own components, you should render an instance of that other component. At worst, you should duplicate the styles within your own component's stylesheet. This is intended to improve maintainability by isolating shared components as a reusable interface, reducing the surface area of similar UI elements by adapting a limited set of common components to support a varied set of use-cases.

It also introduces additional legacy compatibility class names intended to have been removed as part of #19050 (though I also note that this is problematic in the `format-library` from which this code was presumably copied).

Reply:
@aduth This was moved in this PR, but originated in #16897 four months ago. I can see you've opened a PR addressing this so I'm going to close this as resolved.